nestkernel/node_collection.cpp

```diff
@@ -475,18 +481,33 @@ NodeCollectionPrimitive::overlapping( const NodeCollectionPrimitive& rhs ) const
   return ( ( rhs.first_ <= last_ and rhs.first_ >= first_ ) or ( rhs.last_ <= last_ and rhs.last_ >= first_ ) );
 }
 
+bool
+NodeCollectionPrimitive::all_model_ids_same_as_expected_( index expected_model_id ) const
```

Comment:
As commented above, turn into `assert_consistent_model_ids_()` and throw exception from here. Use `Compose()` to build error message telling which node ID causes problem.